test/integration/tree_test.go

```diff
@@ -375,6 +375,72 @@ func TestTree(t *testing.T) {
 		assert.NoError(t, err)
 	})
 
+	t.Run("find correct path after deleting leftNode test", func(t *testing.T) {
+		ctx := context.Background()
+		d1 := document.New(helper.TestDocKey(t))
+		assert.NoError(t, c1.Attach(ctx, d1))
+		d2 := document.New(helper.TestDocKey(t))
+		assert.NoError(t, c2.Attach(ctx, d2))
+
+		assert.NoError(t, d1.Update(func(root *json.Object, p *presence.Presence) error {
+			root.SetNewTree("t", &json.TreeNode{
+				Type: "r",
+				Children: []json.TreeNode{{
+					Type: "p",
+					Children: []json.TreeNode{{
+						Type:  "text",
+						Value: "hello",
+					}},
+				}},
+			})
+
+			return nil
+		}))
+		assert.NoError(t, c1.Sync(ctx))
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><p>hello</p></r>", d1.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><p>hello</p></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, d1.Update(func(root *json.Object, p *presence.Presence) error {
+			root.GetTree("t").EditByPath([]int{0, 3}, []int{0, 3}, &json.TreeNode{
+				Type:  "text",
+				Value: "3",
+			}, 0)
+
+			return nil
+		}))
+
+		assert.NoError(t, c1.Sync(ctx))
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><p>hel3lo</p></r>", d1.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><p>hel3lo</p></r>", d2.Root().GetTree("t").ToXML())
+
+		fromPos, _ := d2.Root().GetTree("t").PathToPos([]int{0, 5})
+		fromParent, fromLeft := d2.Root().GetTree("t").ToTreeNodes(fromPos)
+
+		assert.NoError(t, d2.Update(func(root *json.Object, p *presence.Presence) error {
+			root.GetTree("t").EditByPath([]int{0, 4}, []int{0, 5}, nil, 0)
+			assert.Equal(t, "<r><p>hel3o</p></r>", d2.Root().GetTree("t").ToXML())
+
+			root.GetTree("t").EditByPath([]int{0, 4}, []int{0, 4}, &json.TreeNode{
+				Type:  "text",
+				Value: "m",
+			}, 0)
+			assert.Equal(t, "<r><p>hel3mo</p></r>", d2.Root().GetTree("t").ToXML())
+
+			return nil
+		}))
+		fromPath, err := d2.Root().GetTree("t").ToPath(fromParent, fromLeft)
+		assert.NoError(t, err)
+		assert.Equal(t, []int{0, 5}, fromPath)
+
+		assert.NoError(t, c2.Sync(ctx))
+		assert.NoError(t, c1.Sync(ctx))
+
+		assert.Equal(t, "<r><p>hel3mo</p></r>", d1.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><p>hel3mo</p></r>", d2.Root().GetTree("t").ToXML())
+	})
```

Comment:
It would be nice if the test code was grouped by step with new lines. Could you explain the steps in this test?

Reply:
I will add a description of each step as a comment.